components/raftstore/src/store/fsm/apply.rs

```diff
@@ -190,6 +188,28 @@ impl<C> PendingCmdQueue<C> {
     fn set_conf_change(&mut self, cmd: PendingCmd<C>) {
         self.conf_change = Some(cmd);
     }
+
+    fn push_compact(&mut self, cmd: PendingCmd<C>) {
+        self.compacts.push_front(cmd);
+    }
+
+    fn pop_compact(&mut self, index: u64) -> Option<PendingCmd<C>> {
+        let mut pos: usize = 0;
+        let mut found = false;
+
+        for (i, c) in self.compacts.iter().enumerate() {
```

Comment:
You mean `self.compacts.iter().position(|c| c.index < index)`?

Reply:
Fixed, thanks!